### PR TITLE
fix: Fix links in Announcements

### PR DESCRIPTION
**Pull request by ozandogrultan**
Signed-off-by: Ozan Dogrultan <ozan.dogrultan@deliveryhero.com>

Fixes https://github.com/amundsen-io/amundsen/issues/1875

### Summary of Changes

Previously, when there were links in the Announcement html content, they were overridden by the navigation of the Card component and when they were clicked, announcement list page was shown instead of the link itself.

The changes in the PR introduces the logic the stop bubbling up the click event to the Card component so that the link navigation works as expected.

### Tests

No test change is needed.

### Documentation

No documentation change is needed.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Review by feng-tao — APPROVED** [review]

**Comment by feng-tao** [comment]
cc @Golodhros

**Comment by ozandogrultan** [comment]
@Golodhros could you review this please? 🙏🏼

File: frontend/amundsen_application/static/js/components/Card/index.tsx [hunk 1]
```diff
@@ -44,7 +45,17 @@ const Card: React.FC<CardProps> = ({
...
+        {copy && (
+          <div
+            className="card-copy"
+            role="button"
+            tabIndex={0}
+            onClick={stopEvent}
+            onKeyDown={stopEvent}
+          >
+            {copy}
+          </div>
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Card/index.tsx:57]: This creation of a fake button and adding a stop propagation handler to it seems a bit fishy to me.
Could we achieve the same effect by using CSS pointer-events?
Check in https://css-tricks.com/almanac/properties/p/pointer-events/

**Review by Golodhros — CHANGES_REQUESTED** [review]

File: frontend/amundsen_application/static/js/components/Card/index.tsx (hunk 1 above)
Comment (Golodhros) [reply frontend/amundsen_application/static/js/components/Card/index.tsx:57]: It is something like this: https://stackoverflow.com/a/46707009

Comment (ozandogrultan) [reply frontend/amundsen_application/static/js/components/Card/index.tsx:57]: I've tried this suggestion as well, but it didn't work either.. but I might be doing something wrong, I don't have much experience with `pointer-events`

Another option in my opinion is that eventPropagation can be stopped on the SanitizedHtml level [here](https://github.com/deliveryhero/amundsen/blob/main/frontend/amundsen_application/static/js/components/Announcements/AnnouncementsList/index.tsx#L49)

Comment (ozandogrultan) [reply frontend/amundsen_application/static/js/components/Card/index.tsx:57]: if we stop the propagation in the "html_content" level it also works, but for me it's just another workaround and doesn't solve the problem fully

**Comment by ozandogrultan** [comment]
@Golodhros I've changed the fix strategy slightly with the latest changes. Now the only change is to allow `onclick` handlers to be customized in the sanitized announcement content, so stopping event propagation logic can be set if needed by the content which is passed. Can you please review again?

**Review by Golodhros — APPROVED** [review]
LGTM

**Pull request merged**